kernel/aster-nix/src/fs/utils/dentry.rs

```diff
@@ -147,30 +106,25 @@ impl Dentry {
         DentryFlags::from_bits(flags).unwrap()
     }
 
-    fn is_mountpoint(&self) -> bool {
+    pub fn is_mountpoint(&self) -> bool {
         self.flags().contains(DentryFlags::MOUNTED)
     }
 
-    fn set_mountpoint(&self) {
+    pub fn set_mountpoint(&self) {
```

Comment:
I suggest merging `dentrymnt.rs` into `dentry.rs`. In software engineering, we want to write modular code. The module boundaries should be set to make modules _highly cohesive_ and _loosely coupled_. This goal is not achieved by the current version. Some originally private methods of `Dentry` such as `xx_mountpoint` are made public. New helper methods like `lookup_xx` are added. But the only users of these methods are `DentryMnt`. This shows that the two types are highly coupled.